unstructured_ingest/v2/processes/connectors/fsspec/fsspec.py

```diff
@@ -303,10 +303,14 @@ def precheck(self) -> None:
             raise DestinationConnectionError(f"failed to validate connection: {e}")
 
     def get_upload_path(self, file_data: FileData) -> Path:
-        upload_path = (
-            Path(self.upload_config.path_without_protocol)
-            / file_data.source_identifiers.relative_path
-        )
+        upload_path_name: str
+        if file_data.source_identifiers is not None:
+            upload_path_name = file_data.source_identifiers.relative_path.lstrip("/")
+        elif file_data.local_download_path is not None:
+            upload_path_name = Path(file_data.local_download_path).name
+        else:
+            upload_path_name = file_data.identifier
+        upload_path = Path(self.upload_config.path_without_protocol) / upload_path_name
```

Comment:
@potter-potter @rbiseck3 

We had a short discussion with @mateusz-kuprowski about this solution. Shouldn't `source_identifiers` be required for all connectors? I'm guessing it would not be obvious to come up with a path for every connector (like SQL) but this way we would create a more certain way of defining output paths.